### PR TITLE
[ENH] Add new widget: Marker genes (moved from scOrange)

### LINKED ISSUE #126: [ENH] Marker Genes: link to public data bases.
Replace tables in Marker Genes by new public databases. Retain our old gene marker spreadsheet for backward compatibility. Add a **dropdown menu** to select a data base.

The general structure of Cell marker remains the same; Filtering searches in both the gene names and cell type names. The set of organisms now depends on a selected database. Gene Name matcher functionality can be used automatically on the gene names to produce correct output.

The database files are cached via existing **Database update** funcitonality. They are regularly updated.

Databases:

1. CellMarker (human and mouse)

- Link: http://bio-bigdata.hrbmu.edu.cn/CellMarker/
- Human: http://bio-bigdata.hrbmu.edu.cn/CellMarker/download/Human_cell_markers.txt
- Mouse: http://bio-bigdata.hrbmu.edu.cn/CellMarker/download/Mouse_cell_markers.txt
- Relevant columns: speciesType, geneSymbol, cellName, PMID
- For geneSymbol, entries in parentheses mean different peptides coming from the same gene: ITGAM, CD14, ITGB2, **[FCGR2A, FCGR2B, FCGR2C]**, CD68
- Reference is in column PMID (PubMed ID)

2. Panglao DB (human and mouse)

- Human and mouse: https://panglaodb.se/markers/PanglaoDB_markers_10_May_2019.tsv.gz
- Relevant columns: species, official gene symbol, cell type
- Warning: The human (Hs) and mouse genes (Mm) are intermixed in the table! 
- Reference is PanglaoDB.

### PR DESCRIPTION
**Pull request by JakaKokosar**
##### Issue
Fixes #126

##### Description of changes
Widget moved from
 **orangecontrib.single_cell.widgets.owmarkergenes.OWMarkerGenes** 
to 
**orangecontrib.bioinformatics.widgets.OWMarkerGenes.OWMarkerGenes**.

Additional changes to how we manage the database of marker genes. Markers are hosted here: http://download.biolab.si/datasets/bioinformatics/marker_genes/

Access to this data is possible trough [serverfiles](https://github.com/biolab/serverfiles) package.

##### Includes
- [X] Code changes
- [X] Tests
- [X] Documentation

### CODE REVIEW

**Comment by mstrazar** [comment]
Some remarks:

- CellMarker: Display actual PMID (i.e. the number) rather than "PubMed". This would enable sorting/filtering by reference publication.
- Slow responsiveness when entering text, presumably because the table is large. Can this be improved 
- In Databases update, write “Cell Markers: CellMarker” and “Cell Markers: PanglaoDB”.

Eventually, add the documentation to the widget and mention Database update.

File: orangecontrib/bioinformatics/widgets/OWMarkerGenes.py [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+
+    selected_group: str = settings.Setting('')
+    selected_db_source: str = settings.Setting('')
+    filter_text: str = settings.Setting('')
+    header_state: bytes = settings.Setting(b'')
+
+    settingsHandler = MarkerGroupContextHandler()
+    selected_genes: List[tuple] = settings.ContextSetting([])
+
+    # class Error(widget.OWWidget.Error):
```
Comment (PrimozGodec) [inline orangecontrib/bioinformatics/widgets/OWMarkerGenes.py]: Can you remove that part of the code?

File: orangecontrib/bioinformatics/widgets/OWMarkerGenes.py [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+            idx = -1
+
+        self.db_source_cb.clear()
+        self.db_source_cb.addItems(items)
+
+        if idx != -1:
+            self.db_source_index = idx
+            self.selected_db_source = items[idx]
+        elif items:
+            self.db_source_index = min(max(self.db_source_index, 0), len(items) - 1)
```
Comment (PrimozGodec) [inline orangecontrib/bioinformatics/widgets/OWMarkerGenes.py]: Can you split all lines that are longer than 80 characters in more lines?

**Comment by PrimozGodec** [comment]
I made two suggestions. Can you add the documentation too?

**Comment by PrimozGodec** [comment]
I also wrote an issue with two options to improve the widget. They should be proposed in the separate pull request later. https://github.com/biolab/orange3-bioinformatics/issues/132

File: orangecontrib/bioinformatics/widgets/OWMarkerGenes.py (hunk 1 above)
Comment (JakaKokosar) [reply orangecontrib/bioinformatics/widgets/OWMarkerGenes.py]: Thanks, I've missed this.

File: orangecontrib/bioinformatics/widgets/OWMarkerGenes.py (hunk 2 above)
Comment (JakaKokosar) [reply orangecontrib/bioinformatics/widgets/OWMarkerGenes.py]: The default was 88 :D I've reformated it with line length of 80.

**Comment by JakaKokosar** [comment]
> Some remarks:
> 
> * CellMarker: Display actual PMID (i.e. the number) rather than "PubMed". This would enable sorting/filtering by reference publication.

This is now changed.

> * Slow responsiveness when entering text, presumably because the table is large. Can this be improved?

Filtering should feel smoother now, can you confirm?

> * In Databases update, write “Marker Genes: CellMarker” and “Marker Genes: PanglaoDB”.
> 

Done.

> Eventually, add the documentation to the widget and mention Database update.

Basic documentation is added. We can include more information and graphics after "feature freeze".

**Comment by PrimozGodec** [comment]
Thank you @JakaKokosar I think it is ready to merge now. Martin do you agree with changes.

**Pull request merged**